generate_route.py

```diff
@@ -14,103 +12,147 @@
+class Route():
+  """An object for a route based on the input of a start and ending location.
 
   Attributes:
-      latitude: A float for the latitude of the location in Decimal Degrees
-      longitude A float for the longitude of the location in Decimal Degrees
-  """
-
-  def __init__(self,latitude,longitude):
-    self.latitude = latitude
-    self.longitude = longitude
-    self.altitude = None
-    self.x = None
-    self.y = None
-    self.z = None
-
-  def get_lat_lon_tuple(self):
-    return (self.latitude,self.longitude)
-
-  def get_XYZ(Self):
-    pass
-
-  def __repr__(self):
-    return 'Location(%s, %s, %s)' % (self.latitude, self.longitude, self.altitude)
-
-class Route(object):
-  """An object for a route based on the input of a start and ending location
-
-  Attributes:
-      start_location: a Location object for the start of the route
-      end_location: a Location object for the end of the route
-      route: a list of Location objects for each point on the route
+    start_location: a Location object for the start of the route
+    end_location: a Location object for the end of the route
+    route: a list of Location objects for each point on the route
+    distances: a list of distances between each pair of consecutive
+    locations in meters
+    polyline: an encoded format for the route given by the Maps API
+    for ease of drawing
   """
 
   def __init__(self, start_location, end_location):
     self.start_location = start_location
     self.end_location = end_location
     self.route = []
     self.distances = []
-    self.durations = []
     self.polyline = None
+    self.create_route()
 
   def create_route(self):
-    locations,self.distances,self.durations,self.polyline = request_directions(self.start_location.get_lat_lon_tuple(),self.end_location.get_lat_lon_tuple())
+    """Create a route by requesting from Maps API and then adding altitudes/xyz to each point.
+
+    sets attributes for the class based on API response and then calls
+    add_altitudes() to request elevation data for each point, and then add xyz
+    conversion to each point
+    """
     route = []
-    for location in locations:
+    locations, self.distances, self.polyline = request_directions(self.start_location.get_lat_lon_tuple(), self.end_location.get_lat_lon_tuple())
+    elevations = request_elevations(locations)
+    for location, altitude in zip(locations, elevations):
       latitude = location[0]
       longitude = location[1]
-      route.append(Location(latitude,longitude))
+      route.append(Location(latitude, longitude, altitude))
     self.route = route
-  
-  def add_altitudes(self):
-    locations = []
-    for location in self.route:
-      locations.append(location.get_lat_lon_tuple())
-    elevations = request_elevations(locations)
-    for location,elevation in zip(self.route,elevations):
-      location.altitude = elevation
-  
+    self.start_location = route[0]
+    self.end_location = route[-1]
```

Comment:
The latitude and longitude coordinates are the same, but Location objects initialized with just lat/lon as inputs would have a default altitude whereas route[0] would be a Location object initialized with an accurate altitude. Although, this does overwrite any altitude that was initialized in the input. Now that I think about it, the self.start_location and self.end_location variables aren't that useful since that information is contained in self.route and create_route happens during initialization